### PR TITLE
Adding Code Snippets

### PR DESCRIPTION
**Pull request by superdiana**

### CODE REVIEW

File: sms/send-signed-sms.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+from dotenv import load_dotenv
+
+#Load the environment
+envpath = join(dirname(__file__),'./.env')
+load_dotenv(envpath)
+
+#Init the client
+client = nexmo.Client(
+    key = os.getenv('NEXMO_API_KEY'),
+    secret = os.getenv('NEXMO_API_SECRET'),
```
Comment (slorello89) [inline sms/send-signed-sms.py]: They shouldn't need to add an API Secret if they're signing (the signature acts as the auth)

**Comment by dragonmantank** [comment]
On a side note, Python still need support for all the hashing algorithms, so might be a good idea to fully do this after we add those in.

File: sms/send-signed-sms.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    signature_secret = os.getenv('NEXMO_SIGNATURE_SECRET'),
+    signature_method = 'md5'
```
[the remark this answers is not in the document]
Comment (superdiana) [reply sms/send-signed-sms.py]: Yes, HMAC needs a cryptographic function like md5, sha1 to make the calculations... In other words HMAC depends on a cryptographic functions and other variables like a secret. Is there a specific requirement for either? i just went with md5 as the standard.

File: sms/send-signed-sms.py (hunk 1 above)
Comment (superdiana) [reply sms/send-signed-sms.py]: Took after the other code snippet models. My bad. Cleaning up

**Comment by superdiana** [comment]
To use the async variant of the number insight a change in the `get_advanced_number_insight` Client method is needed. 

For example, proposing:

```python
def get_advanced_number_insight(self, params=None, **kwargs):
    argoparams = params or kwargs
    if "callback" in argoparams:
        return self.get(self.api_host, "/ni/advanced/async/json", params or kwargs)
    else:   
        return self.get(self.api_host, "/ni/advanced/json", params or kwargs)```

I tested with the no callback param and is working too, and it works like a charm. Will submit a pr for that too

**Pull request merged**